storm-core/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -1370,9 +1444,820 @@ public static int toPositive(int number) {
+        return InetAddress.getLocalHost().getCanonicalHostName();
+    }
+
+    private static String memoizedLocalHostnameString = null;
+
+    public static String memoizedLocalHostname () throws UnknownHostException {
+        if (memoizedLocalHostnameString == null) {
+            memoizedLocalHostnameString = localHostname();
+        }
+        return memoizedLocalHostnameString;
+    }
+
+    /**
+     * Gets the storm.local.hostname value, or tries to figure out the local hostname
+     * if it is not set in the config.
+     * @param conf The storm config to read from
+     * @return a string representation of the hostname.
+    */
+    public static String hostname (Map<String, Object> conf) throws UnknownHostException  {
+        if (conf == null) {
+            return memoizedLocalHostname();
+        }
+        Object hostnameString = conf.get(Config.STORM_LOCAL_HOSTNAME);
+        if (hostnameString == null || hostnameString.equals("")) {
+            return memoizedLocalHostname();
+        }
+        return (String)hostnameString;
+    }
+
+    public static String uuid() {
+        return UUID.randomUUID().toString();
+    }
+
+    public static void exitProcess (int val, Object... msg) {
+        StringBuilder errorMessage = new StringBuilder();
+        errorMessage.append("Halting process: ");
+        for (Object oneMessage: msg) {
+            errorMessage.append(oneMessage);
+        }
+        String combinedErrorMessage = errorMessage.toString();
+        LOG.error(combinedErrorMessage, new RuntimeException(combinedErrorMessage));
+        Runtime.getRuntime().exit(val);
+    }
+
+    /**
+     * "{:a 1  :b 2} -> {1 :a  2 :b}"
+     *
+     * Note: Only one key wins if there are duplicate values.
+     *       Which key wins is indeterminate:
+     * "{:a 1  :b 1} -> {1 :a} *or* {1 :b}"
+     */
+    public static <K, V> Map<V, K> simpleReverseMap(Map<K, V> map) {
+        Map<V, K> ret = new HashMap<V, K>();
+        for (Map.Entry<K, V> entry : map.entrySet()) {
+            ret.put(entry.getValue(), entry.getKey());
+        }
+        return ret;
+    }
+
+    /**
+     * "{:a 1 :b 1 :c 2} -> {1 [:a :b] 2 :c}"
+     *
+     * Example usage in java:
+     *  Map<Integer, String> tasks;
+     *  Map<String, List<Integer>> componentTasks = Utils.reverse_map(tasks);
+     *
+     * @param map
+     * @return
+     */
+    public static <K, V> HashMap<V, List<K>> reverseMap(Map<K, V> map) {
+        HashMap<V, List<K>> rtn = new HashMap<V, List<K>>();
+        if (map == null) {
+            return rtn;
+        }
+        for (Entry<K, V> entry : map.entrySet()) {
+            K key = entry.getKey();
+            V val = entry.getValue();
+            List<K> list = rtn.get(val);
+            if (list == null) {
+                list = new ArrayList<K>();
+                rtn.put(entry.getValue(), list);
+            }
+            list.add(key);
+        }
+        return rtn;
+    }
```

Comment:
It is a List because that is what the clojure returned.
```
(defn reverse-map
  "{:a 1 :b 1 :c 2} -> {1 [:a :b] 2 :c}"
  [amap]
  (reduce (fn [m [k v]]
            (let [existing (get m v [])]
              (assoc m v (conj existing k))))
          {} amap))
```